osu.Game/Skinning/Editor/SkinSelectionHandler.cs

```diff
@@ -20,6 +20,89 @@ namespace osu.Game.Skinning.Editor
 {
     public class SkinSelectionHandler : SelectionHandler<ISkinnableDrawable>
     {
+        private const string closest_text = @"Closest";
+
+        /// <summary>
+        /// The hash code of the "Closest" menu item in the anchor point context menu.
+        /// </summary>
+        /// <remarks>Needs only be constant and distinct from any <see cref="Anchor"/> values.</remarks>
+        private static readonly int closest_text_hash = closest_text.GetHashCode();
+
+        private static readonly Dictionary<int, string> anchor_menu_items;
+        private static readonly Dictionary<int, string> origin_menu_items;
+
+        static SkinSelectionHandler()
+        {
+            var anchorMenuSubset = new[]
+            {
+                Anchor.TopLeft,
+                Anchor.TopCentre,
+                Anchor.TopRight,
+                Anchor.CentreLeft,
+                Anchor.Centre,
+                Anchor.CentreRight,
+                Anchor.BottomLeft,
+                Anchor.BottomCentre,
+                Anchor.BottomRight,
+            };
+
+            var texts = anchorMenuSubset.Select(a => a.ToString()).Prepend(closest_text).ToArray();
+            var hashes = anchorMenuSubset.Cast<int>().Prepend(closest_text_hash).ToArray();
+
+            var anchorPairs = hashes.Zip(texts, (k, v) => new KeyValuePair<int, string>(k, v)).ToArray();
+            anchor_menu_items = new Dictionary<int, string>(anchorPairs);
+            var originPairs = anchorPairs.Where(pair => pair.Key != closest_text_hash);
+            origin_menu_items = new Dictionary<int, string>(originPairs);
+        }
+
+        private Anchor getClosestAnchorForDrawable(Drawable drawable)
+        {
+            var parent = drawable.Parent;
+            if (parent == null)
+                return drawable.Anchor;
+
+            // If there is a better way to get this information, let me know. Was taken from LogoTrackingContainer.ComputeLogoTrackingPosition
+            // I tried a lot of different things, such as just using Position / ChildSize, but none of them worked properly.
```

Comment:
This comment needs to disappear before this is merged - first-person comments are not the right way to do this. For this type of commentary, you can use self-posted review comments on the actual github PR.